dist/radar_client.js

```diff
@@ -42,6 +42,233 @@ instance.Backoff = Backoff;
 
 module.exports = instance;
 },
+"lib/message_request.js": function(module, exports, require){var logger = require('minilog')('message:request');
+
+var opTable = {
+  control: ['nameSync'],
+  message: ['publish', 'subscribe', 'sync', 'unsubscribe'],
+  presence: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  status: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  stream: ['get', 'push', 'subscribe', 'sync']
+};
+
+var Request = function (operation, scope) {
+  this.message = {
+    op: operation,
+    to: scope
+  };
+
+  if (!this._isValid()) {
+    throw new Error('Invalid request message.');
+  }
+};
+
+Request.buildGet = function (scope, options) {
+  return new Request('get', scope).setOptions(options);
+};
+
+Request.buildPublish = function (scope, value) {
+  var request = new Request('publish', scope);
+  if (value) {
+    request.message.value = value;
+
+    return request;
+  }
+};
+
+Request.buildPush = function (scope, resource, action, value) {
+  var request = new Request('push', scope);
+  if (resource && action && value) {
+    request.message.resource = resource;
+    request.message.action = action;
+    request.message.value = value;
+
+    return request;
+  } else {
+    throw new Error('Invalid push request message.');
+  }
+};
+
+Request.buildNameSync = function (scope, options) {
+  return new Request('nameSync', scope).setOptions(options);
+};
+
+Request.buildSet = function (scope, value, clientData, key, userType) {
+  var request = new Request('set', scope);
+  if (value) {
+    request.message.value = value;
+    request.message.key = key;
```

Comment:
So, a suggestion: 

	Request.buildPush = function (scope, resource, action, value) {
	  var request;
	  
	  if (!resource  || !action) {
	  	throw new Error('Invalid push request message.');
	  }

	  request = new Request('push', scope);
	  request.setValue(value);
	  request.setAttr('resource', resource);
	  request.setAttr('action', action);

	  return request;
	};


@vanchi-zendesk what do you think?